### PR TITLE
feat(#1355): liquidity provision dashboard details

### LINKED ISSUE #1355: Liquidity / market details
**User Story** 
As a liquidity provider 
I want to be able to see the current spread of liquidity, fee levels, margin requirement, LP share and APY 
So that I can understand the spread of liquidity provided right now and where I might fit in 

**Acceptance Criteria**

- There is a way to interact with a market to view secondary information about the liquidity provided and visually indicated i.e. mouseover, arrow icon
- I can see per LP - party address, time in market, GALPS, Committed liquidity, margin req., 24h fees, fee level, APY
- There is a way to go back to the market list

Screenshot 2022-09-20 at 15 35 48

Dashboard sketch for reference https://miro.com/app/board/uXjVOs_PBKk=/?moveToWidget=3458764531971937952&cot=14

Visual styles (only) can be used from this early version of a design https://www.figma.com/file/qgCd4r61Z1r3tUkxKehhmu/Website-Design-Updates?node-id=399%3A20530

### PR DESCRIPTION
**Pull request by malinantonsson**
# Related issues 🔗

Closes #1355 

# Description ℹ️

This PR adds the Liquidity provision "second level" information (aka details). When a user is viewing the dashboard he can click on a row and access the information in a new tab, with a share-able url (`markets/:marketId`). The details page has the key information for the market plus information for each liquidity provider. 

# Demo 📺

Dashboard
dashboard

Details page
details

### CODE REVIEW

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  return (
+    <AsyncRenderer loading={loading} error={error} data={localData}>
+      <div
+        className="grow w-full"
+        style={{ minHeight: 500, overflow: 'hidden' }}
+      >
+        <Grid
+          gridOptions={{
+            onRowClicked: ({ data }: RowClickedEvent) => {
+              window.open(
```
Comment (malinantonsson) [inline apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx:85]: Not loving this but I couldn't find another way to do it. Any ideas please let me know!

File: apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  fee: string;
+  commitmentAmount: number;
+}
+
+interface settlementAsset {
+  symbol?: string;
+  decimals?: number;
+}
+
+const candlesProvider = makeDerivedDataProvider(
```
Comment (malinantonsson) [inline apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx:38]: @mattrussell36 Please let me know if this should be done differently - the existing hook for a single market in `libs/liquidity` does not include candles, and I can't find any hooks for the candles. Both `apps/trading` and `apps/console-lite` imports the provider directly.

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { t } from '@vegaprotocol/react-helpers';
+
+import { Intro } from './intro';
+import { MarketList } from './market-list';
+
+export function Dashboard() {
+  return (
+    <>
+      <div className="px-16 pt-20 pb-12 bg-[#F0F0F0]">
```
Comment (malinantonsson) [inline apps/liquidity-provision-dashboard/src/app/components/dashboard/dashboard.tsx:9]: @mattrussell36 some new colours are used here, they are part of the styleguide Joelle has put together. Should I add them to a tailwind config or keep using them like this? If so, do I add them to the base config or only for this project?

File: apps/liquidity-provision-dashboard/src/app/components/detail/providers/providers.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    return '-';
+  }
+
+  const MS_IN_HOUR = 1000 * 60 * 60;
+  const created = new Date(value).getTime();
+  const now = new Date().getTime();
+  return `${Math.round(Math.abs(now - created) / MS_IN_HOUR)}h`;
+};
+
+export const Providers = ({
```
Comment (ciaran-) [inline apps/liquidity-provision-dashboard/src/app/components/detail/providers/providers.tsx:23]: This isn't a structural question really - but are we worried about potential confusion between the term `Provider` between Liquidity Providers and React Context Providers?

('No' is a reasonable answer of course!)

Comment (malinantonsson) [reply apps/liquidity-provision-dashboard/src/app/components/detail/providers/providers.tsx:23]: Haha that is a fair question (we also use `dataProviders` for fetching data 😅 ). Do you think I should rename it to LiquidityProviders? Or maybe `Orders`... What do you think? :)

Comment (ciaran-) [reply apps/liquidity-provision-dashboard/src/app/components/detail/providers/providers.tsx:23]: Not sure! Is there much precedent for stuff like this in the project? It's easy enough for us to change in the future anyway, so I don't imagine holding up on this.

Comment (mattrussell36) [reply apps/liquidity-provision-dashboard/src/app/components/detail/providers/providers.tsx:23]: I'm not too worried about the confusion around this, though perhaps a slightly more descriptive name might be suitable? Perhaps LPProvidersGrid?

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx (hunk 1 above)
Comment (mattrussell36) [reply apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx:81]: so for normal navigation you'd normally use the `navigate` function from react router but if you want a new tab I think window.open is the way to go

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/dashboard.tsx (hunk 3 above)
Comment (mattrussell36) [reply apps/liquidity-provision-dashboard/src/app/components/dashboard/dashboard.tsx]: Definitely add to config and for now I'd say keep it in this projects config only

File: apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx (hunk 2 above)
Comment (mattrussell36) [reply apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx]: Looks like this code has changed?

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    <>
+      <span style={{ lineHeight: '12px' }}>{value}</span>
+      <span style={{ lineHeight: '12px' }}>
+        {data?.tradableInstrument?.instrument?.product?.settlementAsset?.symbol}
+      </span>
+    </>
+  );
+};
+
+const healthCellRenderer = ({
```
Comment (mattrussell36) [inline apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx:43]: chore (blocking): This is just a component so lets upper case it to follow convention. If you then can't pass it to ag grid Id wrap it with a function.

```
cellRenderer={(props) => <HealthCell {...props} />}
```

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  value,
+  data,
+}: {
+  value: MarketTradingMode;
+  data: Market;
+}) => {
+  return (
+    <div>
```
Comment (mattrussell36) [inline apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx:51]: chore (non blocking): unnecessary div here?

File: apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            field="tradableInstrument.instrument.name"
+            cellRenderer={marketNameCellRenderer}
+            minWidth={100}
+            flex="1"
+          />
+
+          <AgGridColumn
+            headerName={t('Volume (24h)')}
+            field="dayVolume"
+            cellRenderer={({ value, data }: GroupCellRendererParams) => {
```
Comment (mattrussell36) [inline apps/liquidity-provision-dashboard/src/app/components/dashboard/market-list/market-list.tsx:109]: suggestion (non blocking): I'd suggest using valueFormatter where possible and only use cellRenderer if you really need some extra html

File: apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+}) => {
+  const [isHealthDialogOpen, setIsHealthDialogOpen] = useState(false);
+
+  return (
+    <div>
+      <div className="border border-greys-light-200 rounded-2xl px-2 py-6">
+        <table className="w-full">
+          <thead>
+            <tr
+              className="font-[15px] text-[#626262] text-left font-alpha"
```
Comment (mattrussell36) [inline apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx:47]: suggestion (non blocking): Use the default tw classes if possible? Or add to config

File: apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx (hunk 2 above)
Comment (malinantonsson) [reply apps/liquidity-provision-dashboard/src/app/components/detail/Market/market.tsx]: Yes, this is now in `apps/liquidity-provision-dashboard/src/app/components/detail/last-24h-volume/last-24h-volume.tsx`

**Comment by ciaran-** [comment]
Hi @mattrussell36 I think I'll be picking this up from @malinantonsson  - are there any changes outstanding for this to be approved?

**Comment by ciaran-** [comment]
> Hi @mattrussell36 I think I'll be picking this up from @malinantonsson - are there any changes outstanding for this to be approved?

I see there's a failing cypress test now. I'll look into that.

**Comment by ciaran-** [comment]
A subsequent execution of the tests seems to have been successful: https://github.com/vegaprotocol/frontend-monorepo/actions/runs/3384714537/jobs/5624123456

I’m not familiar enough with the project yet to say if this makes me feel better. Is it possible this PR has introduced some flakiness or not. @mattrussell36 is there anything else to be done here? (Sorry if you're not the right person to tag, but I see you were reviewing this previously)

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**